src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -120,19 +106,130 @@ def put(self, args):
     def get(self):
         return self.output_queue.get(timeout=1)
 
-    def format_current_process(self, line_number: int, is_completed=False):
-        process_name = self.process.name if self.process else None
-        process_pid = self.process.pid if self.process else None
-        if is_completed:
-            bulk_logger.info(
-                f"Process name: {process_name}, Process id: {process_pid}, Line number: {line_number} completed."
-            )
-        else:
-            bulk_logger.info(
-                f"Process name: {process_name}, Process id: {process_pid}, Line number: {line_number} start execution."
-            )
 
-        return f"Process name({process_name})-Process id({process_pid})-Line number({line_number})"
+def format_current_process(process_name, pid, line_number: int, is_completed=False):
+    if is_completed:
+        bulk_logger.info(
+            f"Process name: {process_name}, Process id: {pid}, Line number: {line_number} completed."
+        )
+    else:
+        bulk_logger.info(
+            f"Process name: {process_name}, Process id: {pid}, Line number: {line_number} start execution."
+        )
+
+    return f"Process name({process_name})-Process id({pid})-Line number({line_number})"
+
+
+def fork_processes_manager(
```

Comment:
If it's convenient, please add some comment to explain the current process model, and the reason of the current design.